lib/stream.js

```diff
@@ -87,7 +87,7 @@ if (version[0] === 0 && version[1] < 12) {
 
   if (!Stream._typedArrayToBuffer) {
     Stream._typedArrayToBuffer = function _typedArrayToBuffer(chunk) {
-      return Buffer.prototype.slice.call(chunk);
+      return Buffer.from(chunk.buffer, chunk.byteOffset, chunk.byteLength);
```

Comment:
the `Buffer.prototype.slice()` would have made a different Buffer. With `Buffer.from()`, the underlying memory will still be shared. Will that be a problem?